File: L13GoldmanSachsEasy/KDifferentPairsArray.java
// import java.util.*;
// public class KDifferentPairsArray {
//     public static void main(String[] args) {
//         try (Scanner sc = new Scanner(System.in)) {
//             int n = sc.nextInt();
//             int k = sc.nextInt();
//             int[] arr = new int[n];
            
//             for (int i = 0; i < n; i++) {
//                 arr[i] = sc.nextInt();
//             }
            
//             Map<Integer, Integer> m = new HashMap<>();
//             for (int x : arr) {
//                 m.put(x, m.getOrDefault(x, 0) + 1);
//             }
            
//             int ans = 0;
//             for (Integer x : m.keySet()) {
//                 if ((k == 0 && m.get(x) > 1) || (k != 0 && m.containsKey(x + k))) {
//                     ans++;
//                 }
//             }
            
//             System.out.print(ans);
//         }
//     }
// }













import java.util.*;

public class KDifferentPairsArray {

    public static void main(String[] args) {
        // Define array and k manually
        int[] arr = {1, 5, 3, 4, 2};  // Example array
        int k = 2;  // Example difference

        // Store occurrences of elements in a HashMap
        Map<Integer, Integer> m = new HashMap<>();
        for (int x : arr) {
            m.put(x, m.getOrDefault(x, 0) + 1);
        }

        int ans = 0;
        for (Integer x : m.keySet()) {
            if ((k == 0 && m.get(x) > 1) || (k != 0 && m.containsKey(x + k))) {
                ans++;
            }
        }

        System.out.print(ans);
    }
}

File: L13GoldmanSachsEasy/KDifferentPairsArray.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

int main()
{
    // Predefined array and difference k
    int arr[] = {1, 5, 3, 4, 2}; // Example array
    int n = sizeof(arr) / sizeof(arr[0]);
    int k = 2; // Example difference value

    // Store occurrences of elements in an unordered_map (HashMap in C++)
    unordered_map<int, int> m;
    for (int i = 0; i < n; i++)
    {
        m[arr[i]]++;
    }

    int ans = 0;
    for (auto &pair : m)
    {
        int x = pair.first;
        if ((k == 0 && m[x] > 1) || (k != 0 && m.count(x + k)))
        {
            ans++;
        }
    }

    cout << "Number of unique pairs with difference " << k << ": " << ans << endl;
    return 0;
}
